sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotThrowFromDestructors.CSharp9.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    record Program
+    {
+        public void Foo() => throw new Exception(); // Compliant
+
+        public Program()
+        {
+            throw new Exception(); // Compliant, ctor covered by S3877
+        }
+
+        ~Program()
+        {
+            throw new Exception(); // Noncompliant {{Remove this 'throw' statement.}}
+//          ^^^^^^^^^^^^^^^^^^^^^^
+
+            void Inner()
+            {
+                throw new Exception(); // Noncompliant, C# 7+
```

Comment:
_Nitpick:_ We don't need this comment since we're in CSharp9 file.
```suggestion
                throw new Exception(); // Noncompliant
```